src/fusor/fusion_matching.py

```diff
@@ -0,0 +1,287 @@
+"""Module for matching assayed fusions against categorical fusions"""
+
+import pickle
+from abc import ABC, abstractmethod
+from pathlib import Path
+
+from civicpy import civic
+
+from fusor.harvester import CIVICHarvester
+from fusor.models import (
+    AssayedFusion,
+    CategoricalFusion,
+    GeneElement,
+    LinkerElement,
+    MultiplePossibleGenesElement,
+    TranscriptSegmentElement,
+    UnknownGeneElement,
+)
+from fusor.translator import Translator
+
+
+class CategoricalFusionSource(ABC):
+    """ABC for CategoricalFusion sources"""
+
+    translator: Translator
+
+    @abstractmethod
+    async def save_categorical_fusions(self, output_dir: Path) -> None:
+        """Save CategoricalFusion objects in a cache
+
+        :param output: A path to the cache
+        :return None
+        """
+
+    @abstractmethod
+    def load_categorical_fusions(self, path: Path) -> list[CategoricalFusion]:
+        """Load in cache of CategoricalFusion objects
+
+        :param path: A path to the cache
+        :return A list of Categorical fusions
+        """
+
+
+class CIVICCategoricalFusions(CategoricalFusionSource):
+    """Class for loading and generating cache of CIViC Categorical Fusions"""
+
+    def __init__(self, translator: Translator):
```

Comment:
Consider a couple things:

* Loading cache: https://github.com/griffithlab/civicpy/blob/master/civicpy/civic.py#L147 
* Creating an instance variable for categorical_variants (split out save_categorical_fusions method to support this)